src/Symfony/Component/VarDumper/Caster/VirtualHookedPropertyStub.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * Represents a hooked property.
+ *
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+class VirtualHookedPropertyStub extends Stub
+{
+    public string $propertyType;
+
+    public function __construct(\ReflectionProperty $reflector)
+    {
+        if (\PHP_VERSION_ID < 80400) {
+            throw new \LogicException('Virtual hooked properties are only supported from PHP 8.4.');
+        }
+
+        $this->propertyType = $reflector->hasType() ? $reflector->getType() : '';
```

Comment:
```suggestion
        $this->propertyType = $reflector->hasType() ? (string) $reflector->getType() : null;
```

Shall we use `null` for "no value" instead of an empty string?

Reply:
I agree